### PR TITLE
[WIP] Creating a ROS 2.0 drone_demo container

### PR DESCRIPTION
**Pull request by ahcorde**
This is a working docker container but I still need to work in some parts:

 - [x] remove personal repos 
 - [x] mavlink
 - [x] rviz_aerial_plugins
```
To compile mavlink I have cloned two repos:
   - git clone https://github.com/mavlink/mavlink --recursive 
   - git clone https://github.com/mavlink/mavlink-gbp-release
 
Then I copy all the files inside the patch folder
   mv mavlink-gbp-release/patch/* mavlink

But the file package.xml is from ROS 1. So I have included the right package.xml in this repo. 
```
Does make sense to create a PR in mavlink-gbp-release for Dashing?

### CODE REVIEW

**Comment by ahcorde** [comment]
Maybe we should create a repo in the OSRF organization called `rviz_aerial_plugins`. For now I have this repo in [my personal github](https://github.com/ahcorde/drone_demo/tree/rviz_aerial_plugins). I have restarted the history, we only need to push it in the right repository

**Comment by ahcorde** [comment]
there is no offical docker for Eloquent yet, but I have used the container osrf/ros2:devel to create a version for Eloquent which is in the [branch](https://github.com/osrf/drone_demo/tree/ahcorde/docker_deploy/ros2_eloquent)

**Comment by ahcorde** [comment]
moved `rviz_aerial_plugins` to its [own repo](https://github.com/osrf/rviz_aerial_plugins)

**Comment by tfoote** [comment]
I created a repo for the rviz_aerial_plugins to land in: https://github.com/osrf/rviz_aerial_plugins

And I've reviewed the ROS 2 drone_demo PR. 

I'll try to take a closer look at mavlink soon, but I think it makes sense to submit a PR upstream for ROS 2 support.

We can get the two other prs cleaned up we a few comments from me and then this will be ready to merge. I tried to build it all tonight, but ran out of disk space and time to rebuild it tonight.

**Comment by tfoote** [comment]
I got the basics running with a few tweaks in #23

File: Dockerfile [hunk 1]
```diff
@@ -9,38 +9,69 @@ RUN apt-get update \
...
+  && /bin/sh -c 'echo ". /usr/share/gazebo/setup.sh" >> ~/.bashrc'
 
-RUN /bin/sh -c 'echo ". /opt/ros/melodic/setup.bash" >> ~/.bashrc' \
- && /bin/sh -c 'echo ". /usr/share/gazebo/setup.sh" >> ~/.bashrc'
+RUN mkdir  /workspace/drone_demo_ros2/src -p
+WORKDIR /workspace/drone_demo_ros2/src
+RUN git clone https://github.com/ahcorde/drone_demo.git -b ahcorde/ros2
+RUN git clone https://github.com/ahcorde/sitl_gazebo.git -b ahcorde/ros2 --recursive
+RUN git clone https://github.com/osrf/uav_testing.git -b ahcorde/ros2
+RUN git clone https://github.com/osrf/drone_demo.git -b rviz_aerial_plugins rviz_aerial_plugins
```
Comment (tfoote) [inline Dockerfile]: This should be updated to the new standalone repo.

Comment (ahcorde) [reply Dockerfile]: In order to change this repo `https://github.com/ahcorde/drone_demo.git -b ahcorde/ros2` we need to work into merge this https://github.com/osrf/drone_demo/pull/18

The others two I changed it https://github.com/osrf/drone_demo/pull/23/commits/0174c0b1eec8d1119a517a24b5821061e5af5adc

**Comment by tfoote** [comment]
@ahcorde Can you update the branches now that sitl_gazebo PRs are merged?

**Comment by ahcorde** [comment]
Sorry I messed up with one "git remote".

Could please try it again?

**Comment by tfoote** [comment]
[ ] I'm unable to command typhoon from rviz2 to arm and take off, but was able to give it a goal position and goto

 [ ] The above is likely related to warnings about action server not available seen in the console output.

 [ ] In rviz the model appears to be stuck to marker in rviz, when I drag the goal marker the model moves with it.

Typhoon console snippet:
<details>
gzserver-2] [Err] [ModelDatabase.cc:414] Unable to download model[model://mavlink_sitl_gazebo/models/typhoon_h480/meshes/cgo3_vertical_arm_remeshed_v1.stl]
[gzserver-2] [Err] [SystemPaths.cc:412] File or path does not exist[""]
[gzserver-2] [Err] [ModelDatabase.cc:414] Unable to download model[model://mavlink_sitl_gazebo/models/typhoon_h480/meshes/cgo3_horizontal_arm_remeshed_v1.stl]
[gzserver-2] [Err] [SystemPaths.cc:412] File or path does not exist[""]
[gzserver-2] [Err] [Visual.cc:2824] No mesh specified
[gzserver-2] [Err] [ModelDatabase.cc:414] Unable to download model[model://mavlink_sitl_gazebo/models/typhoon_h480/meshes/cgo3_horizontal_arm_remeshed_v1.stl]
[gzserver-2] [Err] [SystemPaths.cc:412] File or path does not exist[""]
[gzserver-2] [Err] [ModelDatabase.cc:414] Unable to download model[model://mavlink_sitl_gazebo/models/typhoon_h480/meshes/cgo3_mount_remeshed_v1.stl]
[gzserver-2] [Err] [SystemPaths.cc:412] File or path does not exist[""]
[gzserver-2] [Err] [Visual.cc:2824] No mesh specified
[gzserver-2] [Err] [ModelDatabase.cc:414] Unable to download model[model://mavlink_sitl_gazebo/models/typhoon_h480/meshes/cgo3_mount_remeshed_v1.stl]
[gzserver-2] [Err] [SystemPaths.cc:412] File or path does not exist[""]
[rviz2-4] [ERROR] [rviz2]: Action server not available after waiting
[rviz2-4] [ERROR] [rviz2]: Action server not available after waiting
[qgroundcontrol-start.sh-5] 2.50047 2.5 2.52947 0.029
[rviz2-4] [ERROR] [rviz2]: Action server not available after waiting
</details>

 [ ] Iris doesn't render in gazebo it renders ok in gazebo

Iris console snippets
<details>
2] [Err] [Visual.cc:2824] No mesh specified
[gzserver-2] [Err] [ModelDatabase.cc:414] Unable to download model[model://mavlink_sitl_gazebo/models/rotors_description/meshes/iris_prop_cw.dae]
[gzserver-2] [Err] [SystemPaths.cc:412] File or path does not exist[""]
[gzserver-2] [Err] [ModelDatabase.cc:414] Unable to download model[model://mavlink_sitl_gazebo/models/rotors_description/meshes/iris_prop_cw.dae]
[gzserver-2] [Err] [SystemPaths.cc:412] File or path does not exist[""]
[gzserver-2] [Err] [Visual.cc:2824] No mesh specified
[gzserver-2] [Err] [ModelDatabase.cc:414] Unable to download model[model://mavlink_sitl_gazebo/models/rotors_description/meshes/iris_prop_cw.dae]
[gzserver-2] [Err] [SystemPaths.cc:412] File or path does not exist[""]
[gzserver-2] [Err] [ModelDatabase.cc:414] Unable to download model[model://mavlink_sitl_gazebo/models/rotors_description/meshes/iris.stl]
[gzserver-2] [Err] [SystemPaths.cc:412] File or path does not exist[""]
[gzserver-2] [Err] [Visual.cc:2824] No mesh specified
[gzserver-2] [Err] [ModelDatabase.cc:414] Unable to download model[model://mavlink_sitl_gazebo/models/rotors_description/meshes/iris.stl]
[gzserver-2] [Err] [SystemPaths.cc:412] File or path does not exist[""]
</details>


 [ ] Plane has the label "Quadcopter" otherwise looks good. It renders in gazebo

**Comment by ahcorde** [comment]
> I'm unable to command typhoon from rviz2 to arm and take off, but was able to give it a goal position and goto

Sometimes you need to wait some seconds to be able to arm and takeoff the drone. Check QGroundControl for know when the drone is available

>  The above is likely related to warnings about action server not available seen in the console output.

This is probably because not all the nodes are been launched. In particular the `drone_node` which is node that create the actions

> In rviz the model appears to be stuck to marker in rviz, when I drag the goal marker the model moves with it.

Can you provide a small video or gif about what is going on?

> [Err] [Visual.cc:2824] No mesh specified [gzserver-2] [Err] [ModelDatabase.cc:414] Unable to download model[model://mavlink_sitl_gazebo/models/rotors_description/meshes/iris_prop_cw.dae]

Regarding the traces this is quite strange because your system is trying to find the models here:
```
model://mavlink_sitl_gazebo/models
```

But the right url is:

```
package://mavlink_sitl_gazebo/models
```

I'm generating the URDFs in the /tmp folder for example; `/tmp/iris_0.urdf` or `/tmp/plane_1`. Can you check where is the URDF looking for the meshes?

**Pull request merged**